In [2]:
"""
rate= 10000 qp2=qp2*0.1
I frame qp=10
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[0:20]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[0:20]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[0:20]
"""
import numpy as np
B=0.01
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:35.48276845146628, size : 18.185719209558822
time: 1, acc:40.543424381929285, size : 13.929859834558824
time: 2, acc:36.12719704123104, size : 13.980698529411764
time: 3, acc:35.409026987412396, size : 14.108053768382353
time: 4, acc:33.28575246474322, size : 13.963120404411764
time: 5, acc:32.79569850248449, size : 13.803423713235293
time: 6, acc:33.15161256229176, size : 13.834271599264707
time: 7, acc:34.04433026033289, size : 13.848805147058824
time: 8, acc:30.602811364566577, size : 13.920438878676471
time: 9, acc:26.262945287367877, size : 14.085477941176471
time: 10, acc:29.182153589585248, size : 13.987534466911764
time: 11, acc:32.49406534082749, size : 14.084731158088236
time: 12, acc:33.1543108996223, size : 14.283490349264707
time: 13, acc:36.08210227068733, size : 14.215877757352942
time: 14, acc:40.78463386086857, size : 14.115521599264707
time: 15, acc:37.702030294081744, size : 14.145852481617647
time: 16, acc:40.47102928161621, size : 13.952607996323529
ti

In [ ]:
Version1 

In [3]:
"""
rate= 10000 qp2=qp2*0.25
I frame qp=20
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[5:35]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[0:30]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[0:30]
"""
import numpy as np
B=0.001
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:35.151181501500744, size : 22.177159926470587
time: 1, acc:47.374565461102655, size : 16.67124310661765
time: 2, acc:45.66815881168141, size : 17.585018382352942
time: 3, acc:34.843787025002875, size : 17.59133731617647
time: 4, acc:32.758000317741846, size : 17.1728515625
time: 5, acc:39.23648946425494, size : 16.439280790441178
time: 6, acc:40.068620793959674, size : 16.748219209558822
time: 7, acc:43.36775330936207, size : 17.11569393382353
time: 8, acc:37.29956851286047, size : 17.37172564338235
time: 9, acc:32.98625946044922, size : 17.847541360294116
time: 10, acc:35.15593304353602, size : 17.79331341911765
time: 11, acc:35.74846492094152, size : 17.784581801470587
time: 12, acc:41.6744260226979, size : 18.3046875
time: 13, acc:40.86434700909783, size : 18.070197610294116
time: 14, acc:45.86116285885082, size : 17.381031709558822
time: 15, acc:42.1258056865019, size : 18.349666819852942
time: 16, acc:41.00354699527516, size : 18.02688419117647
time: 17, acc:45.273245

In [5]:
"""
qp2=qp2*0.55
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[25:55]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[20:50]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=0.0001
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:54.819701699649585, size : 31.867474724264707
time: 1, acc:66.82811344371123, size : 25.564223345588236
time: 2, acc:69.60384144502528, size : 26.041015625
time: 3, acc:64.53624893637264, size : 25.928825827205884
time: 4, acc:66.31362858940574, size : 25.300838694852942
time: 5, acc:68.07186743792366, size : 25.744427849264707
time: 6, acc:70.03458247465245, size : 25.924000459558822
time: 7, acc:69.56317565020393, size : 26.51694623161765
time: 8, acc:66.92243744345272, size : 27.011316636029413
time: 9, acc:53.6441213944379, size : 26.904698988970587
time: 10, acc:56.499736449297735, size : 26.766773897058822
time: 11, acc:58.46961245817297, size : 27.397633272058822
time: 12, acc:56.69716386234059, size : 27.404411764705884
time: 13, acc:64.98164289137897, size : 28.956916360294116
time: 14, acc:72.68623183755314, size : 28.242876838235293
time: 15, acc:59.52400039224064, size : 27.320599724264707
time: 16, acc:64.60474238676183, size : 27.384995404411764
time: 17, acc

In [6]:
"""
 qp2=qp2*0.75
I frame qp=42
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[37:67]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[32:62]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[25:55]
"""
import numpy as np
B=0.00008
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:75.4403226515826, size : 38.03360523897059
time: 1, acc:77.79929497662712, size : 30.290613511029413
time: 2, acc:78.81897758035099, size : 31.510799632352942
time: 3, acc:73.47911385928883, size : 31.353515625
time: 4, acc:75.13067021089442, size : 30.747242647058822
time: 5, acc:71.39982335707721, size : 30.94973575367647
time: 6, acc:74.76941557491527, size : 30.660041360294116
time: 7, acc:74.89211138556985, size : 31.268152573529413
time: 8, acc:76.1392453137566, size : 31.848862591911764
time: 9, acc:71.7990847194896, size : 31.497759650735293
time: 10, acc:74.15071094737334, size : 31.281364889705884
time: 11, acc:71.56235189998851, size : 32.05905330882353
time: 12, acc:71.57012715059169, size : 32.98655790441177
time: 13, acc:77.46702082016888, size : 34.393727022058826
time: 14, acc:81.98508655323701, size : 34.15814568014706
time: 15, acc:75.76577523175408, size : 33.11758961397059
time: 16, acc:68.06455499985638, size : 32.578986672794116
time: 17, acc:71.05420

In [8]:
"""
qp2=qp2*0.94
I frame qp=85
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[60:90]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
"""
import numpy as np
B=0.000005
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:97.07087067996754, size : 94.88453584558823
time: 1, acc:96.573086345897, size : 79.80618106617646
time: 2, acc:96.43754959106445, size : 81.10633042279412
time: 3, acc:95.44222495135139, size : 82.33880974264706
time: 4, acc:96.53189603020164, size : 79.99437040441177
time: 5, acc:96.59763504477108, size : 81.29164751838235
time: 6, acc:96.55426530277028, size : 81.52814797794117
time: 7, acc:95.90780594769646, size : 83.66509650735294
time: 8, acc:95.1171734753777, size : 84.12838924632354
time: 9, acc:93.70598793029785, size : 85.33645450367646
time: 10, acc:94.5058261646944, size : 85.01734834558823
time: 11, acc:95.61476707458496, size : 85.30440027573529
time: 12, acc:96.00984629462747, size : 87.66871553308823
time: 13, acc:96.55957221984863, size : 90.98627068014706
time: 14, acc:96.68139289407169, size : 87.43049172794117
time: 15, acc:93.8027353847728, size : 87.87120863970588
time: 16, acc:95.93190305373248, size : 84.62827435661765
time: 17, acc:96.465528712553

In [9]:
"""
qp2=qp2*0.94
I frame qp=90
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')70:100]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')70:100]
"""
import numpy as np
B=0.000001
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:97.1851152532241, size : 109.97380514705883
time: 1, acc:96.7404141145594, size : 87.82421875
time: 2, acc:96.51204838472255, size : 89.31927849264706
time: 3, acc:95.42128058040844, size : 90.74385340073529
time: 4, acc:96.59970788394703, size : 88.04612821691177
time: 5, acc:96.72463641447179, size : 89.86207490808823
time: 6, acc:96.52860024396111, size : 89.97259880514706
time: 7, acc:96.09953375423656, size : 91.91044347426471
time: 8, acc:95.22477318258846, size : 91.91526884191177
time: 9, acc:93.8318364760455, size : 92.26976102941177
time: 10, acc:94.79523266063018, size : 92.11575137867646
time: 11, acc:95.55695197161506, size : 92.8609260110294
time: 12, acc:96.2154135984533, size : 95.55474494485294
time: 13, acc:96.69772596920238, size : 99.35431985294117
time: 14, acc:96.65736591114717, size : 95.72570082720588
time: 15, acc:93.85655066546272, size : 94.62890625
time: 16, acc:96.11469717586742, size : 92.32668887867646
time: 17, acc:96.54413391562069, size : 

In [11]:
"""
 qp2=qp2*0.89
I frame qp=75
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[60:90]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[55:85]
"""
import numpy as np
B=1e-05
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.17600676592659, size : 64.3408203125
time: 1, acc:94.51822954065659, size : 53.89361213235294
time: 2, acc:94.53057962305405, size : 54.93612132352941
time: 3, acc:92.80127637526569, size : 55.27297794117647
time: 4, acc:94.93154637953815, size : 53.711741727941174
time: 5, acc:95.28485746944652, size : 54.74557674632353
time: 6, acc:95.17988597645478, size : 54.98489200367647
time: 7, acc:94.11966660443474, size : 56.3961971507353
time: 8, acc:92.36250484690947, size : 56.974781709558826
time: 9, acc:90.49589213202981, size : 57.16015625
time: 10, acc:91.31901684929343, size : 56.817727481617645
time: 11, acc:93.0220463696648, size : 57.54400275735294
time: 12, acc:94.907247318941, size : 59.49213005514706
time: 13, acc:95.2259568607106, size : 61.84547334558823
time: 14, acc:95.27263360864976, size : 59.53021599264706
time: 15, acc:91.62345213048599, size : 59.41561351102941
time: 16, acc:93.97408822003533, size : 57.33737362132353
time: 17, acc:95.32232565038345, siz

In [12]:
"""
 qp2=qp2*0.72
I frame qp=42
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[38:68]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[25:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=6e-05
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:77.53744125366211, size : 38.356100643382355
time: 1, acc:79.74017648135914, size : 30.314453125
time: 2, acc:84.49655981624828, size : 32.021656709558826
time: 3, acc:78.30679276410271, size : 32.0426815257353
time: 4, acc:80.10444641113281, size : 31.319106158088236
time: 5, acc:81.53036622440113, size : 31.925206801470587
time: 6, acc:83.39480231789982, size : 32.13453584558823
time: 7, acc:80.75619024388931, size : 32.028205422794116
time: 8, acc:78.02274647881003, size : 32.83243336397059
time: 9, acc:72.32211617862477, size : 31.98000919117647
time: 10, acc:72.88375461802764, size : 31.732823988970587
time: 11, acc:73.24874541338752, size : 32.12459788602941
time: 12, acc:76.1365946601419, size : 33.30101102941177
time: 13, acc:80.99950622109806, size : 34.549460018382355
time: 14, acc:81.19629411136403, size : 33.004078584558826
time: 15, acc:68.95616475273582, size : 32.556640625
time: 16, acc:80.49964063307819, size : 33.078929227941174
time: 17, acc:80.0842453451

In [13]:
"""
 qp2=qp2*0.72
I frame qp=42
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[38:68]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[25:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=5e-05
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:78.17882650038776, size : 38.47323069852941
time: 1, acc:80.53719296174891, size : 30.367876838235293
time: 2, acc:84.58876890294692, size : 32.075022977941174
time: 3, acc:78.88573758742389, size : 32.125861672794116
time: 4, acc:80.7088431190042, size : 31.391544117647058
time: 5, acc:81.46194850697236, size : 32.008157169117645
time: 6, acc:82.10447535795323, size : 32.21834788602941
time: 7, acc:80.46803754918716, size : 32.9856387867647
time: 8, acc:80.2980899810791, size : 32.15625
time: 9, acc:72.67669789931354, size : 32.07105928308823
time: 10, acc:70.37352674147662, size : 32.72805606617647
time: 11, acc:72.2977834589341, size : 32.38551240808823
time: 12, acc:78.45811002394733, size : 33.28716681985294
time: 13, acc:81.26564306371353, size : 34.656364889705884
time: 14, acc:82.02524746165557, size : 33.154181985294116
time: 15, acc:73.7732915317311, size : 33.1233340992647
time: 16, acc:84.62436058942009, size : 33.43675321691177
time: 17, acc:79.58225081948673,

In [15]:

"""
 qp2=qp2*0.7
I frame qp=42
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[35:65]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[30:60]
"""
import numpy as np
B=4e-05
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:82.96599107630112, size : 38.656881893382355
time: 1, acc:82.92000153485466, size : 31.11069623161765
time: 2, acc:83.33306032068589, size : 31.632697610294116
time: 3, acc:78.25442482443417, size : 31.520335477941178
time: 4, acc:80.42419657987706, size : 30.771599264705884
time: 5, acc:80.66746487337001, size : 31.409926470588236
time: 6, acc:81.36646607342888, size : 31.617302389705884
time: 7, acc:79.73924524643841, size : 32.301413143382355
time: 8, acc:77.6191655327292, size : 32.89114200367647
time: 9, acc:71.63720691905303, size : 32.619944852941174
time: 10, acc:72.05926109762753, size : 32.342888327205884
time: 11, acc:75.77883215511547, size : 33.07398897058823
time: 12, acc:82.43974236881031, size : 34.27004825367647
time: 13, acc:85.17410895403694, size : 35.5080997242647
time: 14, acc:85.23904856513528, size : 34.30997242647059
time: 15, acc:79.62036693797393, size : 34.08588005514706
time: 16, acc:81.7974427167107, size : 33.147805606617645
time: 17, acc:86.

In [17]:

"""
 qp2=qp2*0.8
I frame qp=55
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[35:65]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[30:60]
"""
import numpy as np
B=3e-05
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:85.8921022976146, size : 41.52292049632353
time: 1, acc:85.5933525983025, size : 32.043830422794116
time: 2, acc:87.02062719008502, size : 34.06927849264706
time: 3, acc:82.28297794566436, size : 34.1142578125
time: 4, acc:84.63567565469181, size : 33.372931985294116
time: 5, acc:84.95554363026338, size : 34.048196231617645
time: 6, acc:86.10506899216595, size : 34.28957950367647
time: 7, acc:83.18860110114602, size : 35.08616727941177
time: 8, acc:82.12628925547881, size : 34.069680606617645
time: 9, acc:75.72519638959099, size : 35.17066865808823
time: 10, acc:75.04611856797162, size : 35.139763327205884
time: 11, acc:79.34499347911162, size : 34.54451976102941
time: 12, acc:81.62812625660615, size : 35.400562959558826
time: 13, acc:85.94439450432273, size : 37.06439568014706
time: 14, acc:87.97822840073529, size : 36.33536305147059
time: 15, acc:78.83525175206802, size : 35.5202205882353
time: 16, acc:87.20610001507927, size : 35.67779181985294
time: 17, acc:78.17726976

In [12]:

"""
 qp2=qp2*0.8
I frame qp=55
   self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[40:70]
"""
import numpy as np
B=0.01
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:35.48276845146628, size : 18.185719209558822
time: 1, acc:40.543424381929285, size : 13.929859834558824
time: 2, acc:36.12719704123104, size : 13.980698529411764
time: 3, acc:35.409026987412396, size : 14.108053768382353
time: 4, acc:33.28575246474322, size : 13.963120404411764
time: 5, acc:32.79569850248449, size : 13.803423713235293
time: 6, acc:33.15161256229176, size : 13.834271599264707
time: 7, acc:34.04433026033289, size : 13.848805147058824
time: 8, acc:30.602811364566577, size : 13.920438878676471
time: 9, acc:26.262945287367877, size : 14.085477941176471
time: 10, acc:29.182153589585248, size : 13.987534466911764
time: 11, acc:32.49406534082749, size : 14.084731158088236
time: 12, acc:33.1543108996223, size : 14.283490349264707
time: 13, acc:36.08210227068733, size : 14.215877757352942
time: 14, acc:40.78463386086857, size : 14.115521599264707
time: 15, acc:37.702030294081744, size : 14.145852481617647
time: 16, acc:40.47102928161621, size : 13.952607996323529
ti

In [19]:


"""
 qp2=qp2*0.9
I frame qp=70
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[60:90]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[50:80]
"""
import numpy as np
B=8e-06
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.6021505243638, size : 68.92802159926471
time: 1, acc:94.9645715601304, size : 56.64010799632353
time: 2, acc:94.99258714563706, size : 57.721507352941174
time: 3, acc:93.47158319809857, size : 58.0881778492647
time: 4, acc:95.45425527236041, size : 56.50045955882353
time: 5, acc:95.65904280718635, size : 57.46340762867647
time: 6, acc:95.36190593943877, size : 57.76585477941177
time: 7, acc:94.37455569996553, size : 59.30221737132353
time: 8, acc:92.85836219787598, size : 59.8125
time: 9, acc:91.22013765222886, size : 60.118049172794116
time: 10, acc:92.04382054946002, size : 59.699103860294116
time: 11, acc:93.5817045323989, size : 60.4284237132353
time: 12, acc:95.09653203627643, size : 62.5302734375
time: 13, acc:95.51053608165068, size : 64.93330652573529
time: 14, acc:95.28450685388901, size : 62.423023897058826
time: 15, acc:92.25472001468434, size : 62.43606387867647
time: 16, acc:94.27136813893037, size : 60.0947265625
time: 17, acc:95.32397775089039, size : 66.

In [27]:


"""
 qp2=qp2*0.8
I frame qp=75
 self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[40:70]
"""
import numpy as np
B=9e-06
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:91.62060232723461, size : 49.526539522058826
time: 1, acc:90.82790823543773, size : 38.45921415441177
time: 2, acc:90.18637993756462, size : 39.14499080882353
time: 3, acc:87.63756191029267, size : 39.0595703125
time: 4, acc:91.27766665290383, size : 38.11971507352941
time: 5, acc:91.03129330803367, size : 38.905101102941174
time: 6, acc:91.54470107134651, size : 39.1796875
time: 7, acc:89.07055574304917, size : 40.11856617647059
time: 8, acc:84.87772661096909, size : 40.648265165441174
time: 9, acc:80.90126935173484, size : 40.379767922794116
time: 10, acc:81.67940027573529, size : 40.04159007352941
time: 11, acc:85.86326486924115, size : 40.90033318014706
time: 12, acc:91.49283240823185, size : 42.431755514705884
time: 13, acc:92.89794529185576, size : 43.96237362132353
time: 14, acc:92.41603122037999, size : 42.447265625
time: 15, acc:86.58671940074248, size : 42.111672794117645
time: 16, acc:89.89171701319077, size : 40.800896139705884
time: 17, acc:91.74152823055492, 

In [44]:


"""
 qp2=qp2*0.82
I frame qp=75
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[47:77]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[40:70]
"""
import numpy as np
B=8.001e-06
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:92.23120072308708, size : 52.1653262867647
time: 1, acc:92.12322235107422, size : 41.093922334558826
time: 2, acc:91.86215120203354, size : 41.8200252757353
time: 3, acc:89.55198175766888, size : 41.804859834558826
time: 4, acc:92.73285304798799, size : 40.751091452205884
time: 5, acc:93.10878024381749, size : 41.647575827205884
time: 6, acc:92.69217603346881, size : 41.886086856617645
time: 7, acc:91.13706700942096, size : 42.935431985294116
time: 8, acc:87.18014604905072, size : 43.37425321691177
time: 9, acc:83.65450185887954, size : 43.143210018382355
time: 10, acc:84.35495320488425, size : 42.888039981617645
time: 11, acc:87.47862647561466, size : 43.700367647058826
time: 12, acc:92.9223285001867, size : 45.29762178308823
time: 13, acc:93.7979024999282, size : 46.99098115808823
time: 14, acc:93.34495488335105, size : 45.33748851102941
time: 15, acc:87.78770390678855, size : 44.97012867647059
time: 16, acc:90.74990328620461, size : 43.546702665441174
time: 17, acc:92.7

In [3]:


"""
 qp2=qp2*0.95
I frame qp=85
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[70:100]
"""
import numpy as np
B=5e-07
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:97.21406207365148, size : 114.83754595588235
time: 1, acc:96.9974770265467, size : 99.52200137867646
time: 2, acc:96.81797308080337, size : 101.22679227941177
time: 3, acc:95.76778972850127, size : 102.73046875
time: 4, acc:96.84997165904326, size : 99.8511603860294
time: 5, acc:96.90946691176471, size : 102.03372012867646
time: 6, acc:96.8875015483183, size : 102.19755284926471
time: 7, acc:96.39323178459617, size : 104.0384880514706
time: 8, acc:95.59141888337977, size : 104.08559283088235
time: 9, acc:94.29135603063247, size : 104.41980698529412
time: 10, acc:95.14258048113655, size : 104.3939568014706
time: 11, acc:96.02207576527314, size : 105.25775505514706
time: 12, acc:96.44934710334329, size : 108.18870634191177
time: 13, acc:96.86704523423138, size : 112.2250114889706
time: 14, acc:96.98588988360237, size : 108.42698759191177
time: 15, acc:94.51186236213235, size : 106.89148667279412
time: 16, acc:96.43286817214069, size : 104.54360064338235
time: 17, acc:96.7429

In [46]:


"""
 qp2=qp2*0.98
I frame qp=85
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[70:100]
"""
import numpy as np
B=5.001e-07
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:97.59763268863453, size : 166.69468060661765
time: 1, acc:97.49095580157112, size : 152.66745174632354
time: 2, acc:97.31729170855354, size : 154.94640395220588
time: 3, acc:96.3689411387724, size : 158.40854779411765
time: 4, acc:97.18478707706227, size : 154.03406479779412
time: 5, acc:97.38784397349639, size : 156.79296875
time: 6, acc:97.3098306094899, size : 157.03952205882354
time: 7, acc:96.99453466078815, size : 159.56634880514707
time: 8, acc:96.49177158580108, size : 158.94778262867646
time: 9, acc:95.01507422503303, size : 160.36276424632354
time: 10, acc:96.11792003407197, size : 160.68922334558823
time: 11, acc:96.54963156756233, size : 161.73776424632354
time: 12, acc:96.86484617345474, size : 165.62925091911765
time: 13, acc:97.44096363292022, size : 171.06272977941177
time: 14, acc:97.68639452317181, size : 165.98575367647058
time: 15, acc:95.22858227000518, size : 163.64102711397058
time: 16, acc:96.79170496323529, size : 161.1199448529412
time: 17, acc:97

In [47]:


"""
 qp2=qp2*1
I frame qp=85
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[80:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[75:100]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[70:100]
"""
import numpy as np
B=1e-07
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:97.66467599307789, size : 276.1285041360294
time: 1, acc:97.59014915017521, size : 270.43152573529414
time: 2, acc:97.4226811352898, size : 273.78044577205884
time: 3, acc:96.49632678312413, size : 278.97673483455884
time: 4, acc:97.25052609163173, size : 272.87086397058823
time: 5, acc:97.44726068833295, size : 275.4325597426471
time: 6, acc:97.47810363769531, size : 276.50729549632354
time: 7, acc:97.11745206047507, size : 278.5762293198529
time: 8, acc:96.64337214301614, size : 278.89792049632354
time: 9, acc:95.10845016030704, size : 280.22966452205884
time: 10, acc:96.15120887756348, size : 279.8056640625
time: 11, acc:96.6565216288847, size : 281.94353170955884
time: 12, acc:96.95170346428367, size : 288.5403262867647
time: 13, acc:97.58391380310059, size : 295.2681525735294
time: 14, acc:97.77348181780647, size : 289.9143497242647
time: 15, acc:95.44643233804142, size : 283.3925206801471
time: 16, acc:96.92125039942124, size : 281.55066636029414
time: 17, acc:97.067

Version2 
B=1e-07    96.66            ACC:   93.19
B=5.001e-07   96.54    ACC:   93.14
B=5e-07       96.02         ACC:   92.86
B=8.001e-06     91.34   ACC:  89.20
B=9e-06       90.19         ACC:    88.21
B=8e-06   94.24             ACC:    91.60
B=3e-05    83.38             ACC:    81.84
B=4e-05     82.71          ACC:      81.25
B=5e-05     78.96          ACC:      77.24
B=6e-05    78.17          ACC:       76.84
B=1e-05    93.94          ACC:        91.37
B=0.000001     95.74    ACC:     92.71
B=0.000005      95.63    ACC:     92.64
B=0.0001    65.68            ACC :     64.46
B=0.001       40.26           ACC:    38.28
B=0.01       36.75             ACC:   32.75

In [10]:
"""
qp2=qp2*0.55
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[25:55]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[20:50]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=0.0001001
fps=5.0

filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:56.114858739516315, size : 31.909352022058822
time: 1, acc:62.28498010074391, size : 25.554113051470587
time: 2, acc:69.72479539759019, size : 26.023322610294116
time: 3, acc:64.53025761772605, size : 25.898724724264707
time: 4, acc:66.09502960653866, size : 25.33059512867647
time: 5, acc:67.13822028216194, size : 25.7373046875
time: 6, acc:68.45761747921215, size : 25.929974724264707
time: 7, acc:69.03816671932445, size : 26.525218290441178
time: 8, acc:66.02625005385455, size : 27.05629595588235
time: 9, acc:50.94500709982479, size : 26.895392922794116
time: 10, acc:48.108933953677905, size : 26.773380055147058
time: 11, acc:57.362951951868396, size : 27.381146599264707
time: 12, acc:57.46796832365148, size : 27.400965073529413
time: 13, acc:61.9081693537095, size : 28.961339613970587
time: 14, acc:71.71987365273868, size : 28.235409007352942
time: 15, acc:58.50089858560001, size : 27.314740349264707
time: 16, acc:57.30795018813189, size : 27.381146599264707
time: 17, ac

In [9]:
"""
 qp2=qp2*0.72
I frame qp=42
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[38:68]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[25:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=5.001e-05
fps=5.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:77.49995904810288, size : 38.4723690257353
time: 1, acc:78.68538463816924, size : 30.370289522058822
time: 2, acc:84.5323730917538, size : 32.0745059742647
time: 3, acc:78.11779134413776, size : 32.125114889705884
time: 4, acc:80.23134119370404, size : 31.391716452205884
time: 5, acc:80.2781301386216, size : 32.0075252757353
time: 6, acc:81.94416831521427, size : 32.215992647058826
time: 7, acc:79.95709811939912, size : 32.99161305147059
time: 8, acc:79.32563108556411, size : 32.15682444852941
time: 9, acc:68.7020414015826, size : 32.06784237132353
time: 10, acc:65.62120774213005, size : 32.727653952205884
time: 11, acc:71.45207629484288, size : 32.39039522058823
time: 12, acc:74.74030887379365, size : 33.2883731617647
time: 13, acc:81.85671357547535, size : 34.653492647058826
time: 14, acc:79.35254994560691, size : 33.15383731617647
time: 15, acc:70.28761471019072, size : 33.1233340992647
time: 16, acc:78.15964362200569, size : 33.436868106617645
time: 17, acc:79.97632026